### PR TITLE
CONTRIB: Script to check if a pull request was changed after approval

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Optimize review process to check pull request changes with force-push

## How
Checkout pull request to a local repo and compare merge result before/after
Example:
```diff
(base) 17:50 [yosefe@hpchead ucx]$ ./contrib/pr_merge_check.py --pr 7760 -v
github head commit: 6505e1b
github approved commit: a51b518 (by 'yosefe')
initialized empty repo at /tmp/yosefe/pr_check
fetched upstream/master as 03b935e
comparing a51b518 and 6505e1b when merged to master
 - merge of a51b518 to upstream/master is ea72ecc
 - merge of 6505e1b to upstream/master is 667ada5
diff --git a/src/uct/ib/dc/dc_mlx5.h b/src/uct/ib/dc/dc_mlx5.h
index 04fb9af..4d703af 100644
--- a/src/uct/ib/dc/dc_mlx5.h
+++ b/src/uct/ib/dc/dc_mlx5.h
@@ -49,6 +49,7 @@ struct ibv_ravh {
 
 #define UCT_DC_MLX5_IFACE_TXQP_DCI_GET(_iface, _dci, _txqp, _txwq) \
     { \
+        ucs_assert(_dci != UCT_DC_MLX5_EP_NO_DCI); \
         _txqp = &(_iface)->tx.dcis[_dci].txqp; \
         _txwq = &(_iface)->tx.dcis[_dci].txwq; \
     }
diff --git a/src/uct/ib/dc/dc_mlx5_ep.c b/src/uct/ib/dc/dc_mlx5_ep.c
index 7cf40ce..11d1c1f 100644
--- a/src/uct/ib/dc/dc_mlx5_ep.c
+++ b/src/uct/ib/dc/dc_mlx5_ep.c
@@ -1531,10 +1531,11 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
     uct_iface_h tl_iface       = ep->super.super.iface;
     uint8_t dci_index          = ep->dci;
     uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_iface, uct_dc_mlx5_iface_t);
-    uct_rc_txqp_t *txqp        = &iface->tx.dcis[dci_index].txqp;
-    uct_ib_mlx5_txwq_t *txwq   = &iface->tx.dcis[dci_index].txwq;
     uint16_t pi                = ntohs(cqe->wqe_counter);
     uint8_t pool_index;
+    UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
+
+    UCT_DC_MLX5_IFACE_TXQP_DCI_GET(iface, dci_index, txqp, txwq);
 
     ucs_debug("handle failure iface %p, ep %p, dci[%d] qpn 0x%x, status: %s",
               iface, ep, dci_index, txwq->super.qp_num,
removing /tmp/yosefe/pr_check
```

## Note
This script requires Python3, and `shutil` , `git` packages
It's recommended to run in a conda environment where all deps can be installed locally

### CODE REVIEW

**Review by hoopoepg — APPROVED** [review]

File: contrib/pr_merge_check.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python
+#
+# This script checks if a gihub pull request was changed after approval, for
```
Comment (dmitrygx) [inline contrib/pr_merge_check.py:3]: github

File: contrib/pr_merge_check.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+# example by a force-push or merge operation
+# Method:
+#    - Clone the repo to  a temporary working directory.
```
Comment (dmitrygx) [inline contrib/pr_merge_check.py:6]: extra whitespace between `to` and `a`

File: contrib/pr_merge_check.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+import requests
+import getpass
+import shutil
+import sys
+import os
+
+class PRChecker(object):
+    def __init__(self):
+        # Other fields are set by parse_args()
+        self.remote_name = "upstream"
```
Comment (dmitrygx) [inline contrib/pr_merge_check.py:27]: can we make the name of upstream repo configurable?

**Review by dmitrygx — APPROVED** [review]

**Comment by yosefe** [comment]
@dmitrygx fixed

**Review by dmitrygx — APPROVED** [review]

**Pull request merged**